src/robot/libdoc.py

```diff
@@ -172,10 +181,15 @@ def _get_doc_format(self, format):
                                    ['ROBOT', 'TEXT', 'HTML', 'REST'])
 
     def _get_output_format(self, format, output):
-        extension = os.path.splitext(output)[1][1:].upper()
-        default = 'XML:HTML' if extension == 'LIBSPEC' else extension
-        return self._verify_format('Format', format or default,
-                                   ['HTML', 'XML', 'XML:HTML'])
+        extension = os.path.splitext(output)[1][1:]
+        return self._verify_format('Format', format or extension,
+                                   ['HTML', 'XML', 'LIBSPEC'])
+
+    def _get_spec_doc_format(self, spec_doc_format):
+        if not spec_doc_format:
+            return None
```

Comment:
Wouldn't it simplify code elsewhere if this helper would return `RAW` or `HTML` depending on the format if there's no explicit spec doc format? Getting format here would make it easy to raise a validation error if spec doc format is used with HTML outputs as well.